src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -283,6 +288,22 @@ public sealed partial class TokenCache : ITokenCacheInternal
             }
         }
 
+        private Dictionary<string, string> GetExtraTokenParametersFromResponse(IEnumerable<string> extraTokenParams, MsalTokenResponse response)
+        {
+            Dictionary<string, string> extraTokenParamsDict = new Dictionary<string, string>();
```

Comment:
If you check if `extraTokenParams` is not empty first, you can return an empty dictionary (see CollectionHelper for that). This will avoid allocating a new dictionary every time.